Python/Algorithms/data_structures/heap.py

```diff
@@ -0,0 +1,98 @@
+from collections.abc import Iterable
+
+class Heap:
+    def __init__(self, iterable = None, min_heap = True):
+
+        if not isinstance(iterable, (Iterable, type(None) )):
+            raise ValueError('cannot make a heap of a non iterable value')
+        
+        if not isinstance(min_heap, bool):
+            raise ValueError('min_heap should be either True or False only')
+
+        self.heap = []
+        self.min_heap = min_heap
+        if iterable:
+            if min_heap:
+                for i in iterable:
+                    self.insert(i)
```

Comment:
It just provides an option to instantiate the heap with a list

Reply:
Okay, I got it. the `if min_heap` was causing the constructor to not build the heap in case the heap was specified to be a max heap. I have removed it from `__init__`.